170.kraig local

ThiagoParolo opened the pull request:
## Descrição
Remoção de features irrelevantes para o funcionamento. 

## Link da tarefa
[CPDD-160](https://www.notion.so/cpdd/Craig-Bot-funcionando-localmente-1c5c5f6d25f880b19c9eef021044fb87?pvs=4)

## Tipo de mudança
- [ ] Bugfix
- [ ] Nova funcionalidade
- [X] Refatoração
- [ ] Documentação

## Como foi testado?
Colocar as variáveis relevantes no install.config.example e remover o .example do nome do arquivo.
Rodar o install.sh

## Checklist
- [X] Código segue o padrão de estilo do projeto
- [ ] Testes unitários foram adicionados
- [ ] A documentação foi atualizada (se aplicável)

bronzelle committed:
🚧 feat(app): set up container for the application

bronzelle committed:
🚧 WIP: use an env var to set the host on DATABASE_URL env var

bronzelle committed:
🚧 WIP: Break `install.sh` into installation and execution.

ThiagoParolo committed:
secrets

ThiagoParolo committed:
removidas configurações que serão lidadas pelo workflow e comandos irrelevantes

ThiagoParolo committed:
Removidas features do patreon e upload para serviços de terceiros.

ThiagoParolo committed:
(apaguei dryrun sem querer)

ThiagoParolo committed:
Removidas features do patreon e terceiros

ThiagoParolo committed:
.

ThiagoParolo committed:
dryrun

ThiagoParolo committed:
dryrun

ThiagoParolo committed:
.

ThiagoParolo committed:
pausado postgres

ThiagoParolo committed:
pausado postgres

ThiagoParolo committed:
.

ThiagoParolo committed:
.

ThiagoParolo committed:
startswith error

ThiagoParolo committed:
.

ThiagoParolo committed:
separado install de run

bronzelle committed:
🔧 Change `.Dockerfile` and `docker-compose.yml` for widely compatibility

bronzelle committed:
📝 docs: Add instrunctions on how run in a container.

ThiagoParolo committed:
Kraig

EderBorella committed:
feat: added PR template

ThiagoParolo committed:
Install -> Dryrun

ThiagoParolo committed:
Merge branch 'main' of https://github.com/Coletivo-Popular-Design-Desenvolvimento/craig into 170.KraigLocal

ThiagoParolo committed:
PR template preenchido

EderBorella commented on .github/pull_request_template:
```diff
@@ -0,0 +1,20 @@
+## Descrição
```
esse aqui deixa em branco, é só a template -> [original](https://github.com/Coletivo-Popular-Design-Desenvolvimento/checkin-discord-bot-v1/blob/main/.github/PULL_REQUEST_TEMPLATE/pull_request_template.md)

EderBorella commented on apps/bot/src/modules/recorder/recording.ts:
```diff
@@ -13,10 +13,9 @@ import { ButtonStyle, ComponentType } from 'slash-create';
 import type RecorderModule from '.';
-import { UserExtraType, WebappOpCloseReason } from './protocol';
-import { WebappClient } from './webapp';
+import { UserExtraType } from './protocol';
```
UserExtraType é usado?

ThiagoParolo replied:
Julguei que poderia ser útil pra algo no futuro, caso não podemos resolver depois

rluders commented on compose.yaml:
```diff
@@ -0,0 +1,23 @@
+version: '3.8'
```
Version tá deprecated. Pode arrancar fora.

rluders commented on compose.yaml:
```diff
@@ -0,0 +1,23 @@
+    tty: true
+    ports:
+      - "8080:8080"
+    depends_on:
```
Se vc não adicionar health checks o `depends_on` pode as vezes dar problema.

rluders commented on install.config.example:
```diff
@@ -56,6 +56,7 @@ NODE_VERSION="18.18.2"
 DATABASE_NAME="craig"
```
Ter esse arquivo, ao invés do `.env.example` pra mim é extremamente confuso.

bronzelle replied:
Realmente. Mas a aplicação usa esse `install.config` pra gerar  vários `.env`.  Então na hora que fose gerar iria substituir o criado. Imagino que foi essa a estratégia deles.

ThiagoParolo replied:
Como o bronzelle disse. Vamos resolver a questão da árvore que ficou confusa e abrimos um issue pra resolver isso sem quebrar o app.

rluders commented on Containerfile:
```diff
@@ -0,0 +1,53 @@
+FROM debian:bullseye-slim
```
```yaml
# ======================
# Stage 1: Build stage
# ======================
FROM debian:bullseye-slim AS build

RUN echo "deb http://deb.debian.org/debian bullseye main" > /etc/apt/sources.list.d/debian.list

RUN apt-get update && \
  apt-get install --no-install-recommends -y \
  gcc-10 \
  g++-10 \
  python3-setuptools \
  sudo \
  flac \
  curl \
  git \
  build-essential \
  autoconf \
  automake \
  libtool \
  pkg-config \
  gawk \
  wget \
  ca-certificates

# Set environment variables for build
ENV CC=gcc-10
ENV CXX=g++-10
ENV NODE_OPTIONS="--max-old-space-size=4096"

# Install FDK AAC codec
RUN git clone https://github.com/mstorsjo/fdk-aac.git /tmp/fdk-aac && \
  cd /tmp/fdk-aac && autoreconf -fiv && ./configure && make && make install

# Install FDK AAC encoder CLI
RUN git clone https://github.com/nu774/fdkaac.git /tmp/fdkaac && \
  cd /tmp/fdkaac && autoreconf -i && ./configure && make && make install

# Install Node.js
RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash - && \
  apt-get install -y nodejs

WORKDIR /app

COPY . .

RUN chmod +x run.sh && ./install.sh

# ==========================
# Stage 2: Final stage
# ==========================
FROM debian:bullseye-slim

RUN echo "deb http://deb.debian.org/debian bullseye main" > /etc/apt/sources.list.d/debian.list && \
  apt-get update && apt-get install --no-install-recommends -y \
  flac \
  nodejs \
  ca-certificates && \
  rm -rf /var/lib/apt/lists/*

# Copy only what's needed from build stage
COPY --from=build /usr/local /usr/local
COPY --from=build /app /app

WORKDIR /app

ENV NODE_OPTIONS="--max-old-space-size=4096"

EXPOSE 3000

CMD ["./run.sh"]
```

Usar multistage aqui vai ser super positivo pra conseguir uma imagem mais leve no final.

rluders commented on install.sh:
```diff
@@ -11,7 +11,6 @@ APT_DEPENDENCIES=(
   inkscape          # cook
```
Aqui esse arquivo, onde vc tiver:

Aqui a linha que vc quer mudar no `install.sh`:

```sh
source "$craig_dir/install.config"
```
mude para:
```sh
source "$craig_dir/.env"
```

bronzelle replied:
Acho que podemos criar uma issue para modificar isso

ThiagoParolo replied:
Como o bronzelle disse. Vamos resolver a questão da árvore que ficou confusa e abrimos um issue pra resolver isso sem quebrar o app.

bronzelle commented on .github/workflows/Dryrun.yml:
```diff
@@ -72,3 +52,6 @@ jobs:
         run: cd apps/tasks && yarn prisma:generate && yarn build
+        
+      - name: Run install
+        run: ./install.sh
```
Esse script precisar fazer o build da imagem do containeir, não da aplicação.

ThiagoParolo committed:
Update Containerfile

add multistage

ThiagoParolo committed:
Update compose.yaml

removed version, added healthcheck

ThiagoParolo reviewed (commented):
repassar install.sh

ThiagoParolo committed:
Update pull_request_template

template

ThiagoParolo committed:
Update Containerfile

add multistage

ThiagoParolo committed:
Update Dryrun.yml

Editado para gerar a imagem do container, verificar se deve manter o GHCR ou trocar pro docker

ThiagoParolo reviewed (commented):
Falhou pela falta do run.sh, que ta na develop

ThiagoParolo committed:
Update Dryrun.yml

ThiagoParolo committed:
Update Dryrun.yml

ThiagoParolo committed:
Update Dryrun.yml

ThiagoParolo committed:
Update Dryrun.yml

eu n vi q o caps tava on

ThiagoParolo committed:
add run.sh

ThiagoParolo committed:
Merge branch '170.KraigLocal' into develop

ThiagoParolo committed:
update compose

ThiagoParolo committed:
Merge branch 'develop' of https://github.com/Coletivo-Popular-Design-Desenvolvimento/craig into develop

ThiagoParolo committed:
Dryrun

ThiagoParolo committed:
update relevant files

ThiagoParolo committed:
update relevant files

ThiagoParolo committed:
Merge branch '170.KraigLocal' of https://github.com/Coletivo-Popular-Design-Desenvolvimento/craig into 170.KraigLocal
